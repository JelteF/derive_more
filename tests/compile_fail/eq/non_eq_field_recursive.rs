#[derive(derive_more::Eq)]
struct IntOrFloat {
    a: Box<NotEq<Self>>,
    b: Box<NotEq<IntOrFloat>>,
}

impl PartialEq for IntOrFloat {
    fn eq(&self, _: &Self) -> bool {
        unimplemented!()
    }
}

#[derive(PartialEq)]
struct NotEq<T>(f32, T);

fn main() {}
