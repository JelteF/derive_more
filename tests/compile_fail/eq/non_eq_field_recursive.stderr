error[E0277]: the trait bound `NotEq<IntOrFloat>: std::cmp::Eq` is not satisfied
 --> tests/compile_fail/eq/non_eq_field_recursive.rs:1:10
  |
1 | #[derive(derive_more::Eq)]
  |          ^^^^^^^^^^^^^^^ the trait `std::cmp::Eq` is not implemented for `NotEq<IntOrFloat>`
  |
  = help: the trait `std::cmp::Eq` is implemented for `Box<T, A>`
  = note: required for `Box<NotEq<IntOrFloat>>` to implement `std::cmp::Eq`
note: required by a bound in `derive_more::__private::AssertParamIsEq`
 --> src/cmp.rs
  |
  | pub struct AssertParamIsEq<T: Eq + ?Sized> {
  |                               ^^ required by this bound in `AssertParamIsEq`
  = note: this error originates in the derive macro `derive_more::Eq` (in Nightly builds, run with -Z macro-backtrace for more info)
